enterprise_access/apps/subsidy_access_policy/admin/__init__.py

```diff
@@ -436,7 +443,15 @@ def save_model(self, request, obj, form, change):
             return
 
         try:
-            obj.force_redeem()
+            extra_metadata = {
+                key: form.cleaned_data.get(key)
+                for key in FORCED_REDEMPTION_GEAG_KEYS
+            }
+            if extra_metadata:
```

Comment:
Hm, won't this always create a truthy `extra_metadata` variable? If the form didn't have the geag values set, extra_metadata would still be a dict with 3 keys, which is truthy. I *think* what you want is this:
```suggestion
            extra_metadata = {
                key: form.cleaned_data.get(key)
                for key in FORCED_REDEMPTION_GEAG_KEYS
                if form.cleaned_data.get(key)
            }
            if extra_metadata:
```